src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -501,20 +501,31 @@ class Dockerfile extends DefaultTask {
     }
 
     interface ItemJoiner {
-        String join(Map map)
+        String join(Map<String, String> map)
     }
 
     static class MultiItemJoiner implements ItemJoiner {
         @Override
-        String join(Map map) {
-            map.inject([]) { result, entry -> result << "\"$entry.key\"=\"$entry.value\"" }.join(' ')
+        String join(Map<String, String> map) {
+            map.inject([]) { result, entry ->
+                // TODO: clarify if assert should be used even in productive code and if this would be a valid one either
+                assert !entry.key.matches('.*(?: |(?:\r{0,1}\n)).*')
```

Comment:
mhh ... updated the task above... I think, we should not verify anything here, just handle what will lead to an invalid Dockerfile per definition.
If anyone adds "invalid" characters to a name or variable, this might be intended. As long the definition is correct and we don't produce invalid Dockerfiles, we should not care about because it's like a user writes stupid things into this file manually. Else, we have to be accurate enough to also verify if someone passes a pre-build Dockerfile for reuse ... "not my job" ;)

so, what I see:
* we care about whitespaces because they will change the
i.e. doing the same as with values, if any whitespace is included, enwrap it with quotes to be safe 
* we care about empty keys, because they will break the Dockerfile syntax too, because
`environmentVariable "" "bar"`
  => 
  > ENV bar
  <br>
  is invalid, resp. inverts what the user expects (value becomes the name, value is empty (in best case))